main.go

```diff
@@ -181,21 +185,31 @@ func main() {
 	}
 
 	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
-		Scheme:             scheme,
-		MetricsBindAddress: metricsBindAddr,
-		LeaderElection:     enableLeaderElection,
-		LeaderElectionID:   "controller-leader-election-capo",
-		LeaseDuration:      &leaderElectionLeaseDuration,
-		RenewDeadline:      &leaderElectionRenewDeadline,
-		RetryPeriod:        &leaderElectionRetryPeriod,
-		Namespace:          watchNamespace,
-		SyncPeriod:         &syncPeriod,
-		ClientDisableCacheFor: []client.Object{
-			&corev1.ConfigMap{},
-			&corev1.Secret{},
+		Scheme:           scheme,
+		Metrics:          metricsserver.Options{BindAddress: metricsBindAddr},
+		LeaderElection:   enableLeaderElection,
+		LeaderElectionID: "controller-leader-election-capo",
+		LeaseDuration:    &leaderElectionLeaseDuration,
+		RenewDeadline:    &leaderElectionRenewDeadline,
+		RetryPeriod:      &leaderElectionRetryPeriod,
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{watchNamespace: {}},
```

Comment:
Does this work if `watchNamespace == ""`.
Might be good to adapt what CAPI does: https://github.com/kubernetes-sigs/cluster-api/blob/dc64556b0a19e11e14ea3ba28095d64547f1aa12/bootstrap/kubeadm/main.go#L197-L202

Reply:
Very good suggestion, thanks!